src/browser/modules/Sidebar/Sidebar.tsx

```diff
@@ -92,12 +113,48 @@ const Sidebar = ({
         return (
           <>
             <DrawerHeader>Favorites</DrawerHeader>
+            {scriptDraft && (
+              <NewSavedScript
+                onSubmit={input => {
+                  const draft = scriptDraft || ''
+                  if (input === favoriteName(draft)) {
+                    addFavorite(draft)
+                  } else {
+                    const alreadyHasName = draft.startsWith('//')
+                    const replaceName = [
+                      `// ${input}`,
+                      draft.split('\n').slice(1)
+                    ].join('\n')
+
+                    addFavorite(
+                      alreadyHasName ? replaceName : `//${input}\n${draft}`
+                    )
+                  }
+                  resetDraft()
+                }}
+                defaultName={favoriteName(scriptDraft || '')}
+                headerText={'Save as'}
+                onCancel={resetDraft}
+              />
+            )}
             <Favorites />
-            {showStaticScripts && <StaticScripts />}
           </>
         )
       }
     },
+    isRelateAvailable
```

Comment:
We spoke about this. Seems ok to me, there could be other ways to do this but this works nicely.

Reply:
Did it with the spread operator instead so the filter null is not needed.